### PR TITLE
[NTP Search]: More UX tweaks

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/38704

Had a call with @aguscruiz this morning to run over the changes :smile:

[Screencast from 2024-06-06 16-05-00.webm](https://github.com/brave/brave-core/assets/7678024/17dada7c-9efb-4fdf-8dcf-3c7bdd3fbbd5)

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Icons in search results should line up with the search engine in the input
2. Placeholder text should be white
3. Search Engine picker menu should use the same style as the input/results (not white anymore)
4. It should be possible to hide the searchbox when it's closed via a 3 dot menu which only shows up when you hover
5. Urls should be gone from search results now

**Note:** Something seems to have happened with the search results which is stopping more suggestions coming through - I'm struggling to find the issue, but I'm pretty sure one was filed.

### CODE REVIEW

File: components/brave_new_tab_ui/components/search/SearchPlaceholder.tsx [hunk 1]
```diff
@@ -3,32 +3,86 @@
...
+  }
 `
 
 function Swapper() {
   const { open, setOpen } = useSearchContext()
   const [boxPos, setBoxPos] = React.useState(0)
+  const [, setShowSearchBox] = useNewTabPref('showSearchBox')
   return <>
     {!open && <PlaceholderContainer onClick={e => {
+      console.log("Closing", e.nativeEvent.composedPath())
```
Comment (sangwoo108) [inline components/brave_new_tab_ui/components/search/SearchPlaceholder.tsx:62]: nit: Should this be removed?

File: components/brave_new_tab_ui/components/search/EnginePicker.tsx [hunk 2]
```diff
@@ -3,68 +3,133 @@
...
-  return <ButtonMenu data-theme="light" positionStrategy='fixed' isOpen={open} onClose={() => setOpen(false)}>
-    <OpenButton fab kind="plain-faint" slot="anchor-content" onClick={() => setOpen(true)}>
-      <Flex gap={spacing.s} align='center'>
+  const buttonRef = React.useRef<HTMLElement>();
+  const menuRef = React.useRef<HTMLElement>();
+
+  React.useEffect(() => {
+    if (!open) return
+    const handler = (e: MouseEvent) => {
+      console.log(menuRef.current)
```
Comment (sangwoo108) [inline components/brave_new_tab_ui/components/search/EnginePicker.tsx:88]: ditto

**Review by sangwoo108 — APPROVED** [review]
LGTM

**Review by mkarolin — APPROVED** [review]
`strings`++

**Pull request merged**